### PR TITLE
Enable embedders to return logits

### PR DESCRIPTION
**Pull request by fteufel**
So far, embedders only returned embeddings. Logits and crossentropy can also be useful.

First enabled for HyenaDNA. New options for HyenaDNAEmbedder:

```python3
return_logits: bool=False, return_loss: bool=False
```

Other models WIP.

### CODE REVIEW

**Comment by fteufel** [comment]
- Adds caduceus model
- Fixes hyenadna problems
- logits/crossentropy for HyenaDNA, Caduceus, NT, DNABERT2
